core/src/main/scala/cats/ErrorControl.scala

```diff
@@ -0,0 +1,38 @@
+package cats
+
+import cats.data.EitherT
+
+trait ErrorControl[F[_], G[_], E] extends Serializable {
```

Comment:
I like to use abstract types as "output types". It seems to me we might want `G[_]` to be an abstract type here, since we might often want to write algorithms to work in F go through G, but return an F. Then we could accept an implicit `ErrorControl[F, E]` to do that, which might be nice.

Reply:
I like that as well, see my recent issue in parallel, though I'm not 100% sure if it applies here, have to think about it more.

Reply:
I don't think we'd be able to express these https://github.com/typelevel/cats/pull/2231/files#diff-aadaecd5f11de32b7cdacd702c9a5393R38 maybe you know a way?